index.js

```diff
@@ -1,5 +1,49 @@
 'use strict';
-const npmconf = require('npm/lib/config/core');
-const pify = require('pify');
+const path = require('path');
+const makeDir = require('make-dir');
+const Conf = require('./lib/conf');
+const defaults = require('./lib/defaults');
+const types = require('./lib/types');
 
-module.exports = () => pify(npmconf.load)();
+module.exports = opts => {
+	opts = Object.assign({}, opts);
+
+	const conf = new Conf(Object.assign({}, defaults.defaults));
+
+	conf.add(opts, 'cli');
+	conf.addEnv();
+	conf.loadPrefix();
+
+	const projectConf = path.resolve(conf.localPrefix, '.npmrc');
+	const userConf = conf.get('userconfig');
+
+	if (!conf.get('global') && projectConf !== userConf) {
+		conf.addFile(projectConf, 'project');
+	} else {
+		conf.add({}, 'project');
+	}
+
+	conf.addFile(conf.get('userconfig'), 'user');
+
+	if (conf.get('prefix')) {
+		const etc = path.resolve(conf.get('prefix'), 'etc');
+		makeDir.sync(etc);
```

Comment:
Do we wan't to create these directories or should we just skip? I guess `npm` does it to just *work* but we don't have to.